packages/xrpc-server/src/stream/server.ts

```diff
@@ -0,0 +1,59 @@
+import { IncomingMessage } from 'http'
+import { WebSocketServer, ServerOptions, WebSocket } from 'ws'
+import { ErrorFrame, Frame } from './frames'
+import logger from './logger'
+
+export class XrpcStreamServer {
+  wss: WebSocketServer
+  constructor(opts: ServerOptions & { handler: Handler }) {
+    const { handler, ...serverOpts } = opts
+    this.wss = new WebSocketServer(serverOpts)
+    this.wss.on('connection', async (socket, req) => {
+      socket.on('error', (err) => logger.error(err, 'websocket error'))
```

Comment:
just out of curiosity, what happens on an error here? does the socket keep going? does it terminate? is there any instance in which _we_ should be terminating?

Reply:
Ooooh good call.  Since this is a `net.Socket`, it should close itself immediately after the error, so I think we're alright.  https://nodejs.org/api/net.html#event-error_1

Reply:
Oops, this was correct but for the wrong reason. When double-checking I see that it's _not_ an extension of net.Socket as I thought, but luckily it does have the same behavior here 👍 https://github.com/websockets/ws/issues/1823#issuecomment-740056036

Reply:
awesome thanks for the check 🙏